### PR TITLE
Avoid exporting malloc/free in all pthreads builds. NFC

### PR DESCRIPTION
**Pull request by sbc100**
The JS code for pthread_create does actually depend on malloc or free except in the case of OFFSCREENCANVASES_TO_PTHREAD.

### CODE REVIEW

File: src/library_pthread.js [hunk 1]
```diff
@@ -660,7 +660,11 @@ var LibraryPThread = {
   // allocations from __pthread_create_js we could also remove this.
   __pthread_create_js__noleakcheck: true,
   __pthread_create_js__sig: 'iiiii',
-  __pthread_create_js__deps: ['$spawnThread', 'pthread_self', '$pthreadCreateProxied'],
+  __pthread_create_js__deps: ['$spawnThread', 'pthread_self', '$pthreadCreateProxied',
+#if OFFSCREENCANVAS_SUPPORT
+    'malloc',
```
Comment (kripken) [inline src/library_pthread.js:665]: Doesn't this need to be in deps_info? That is, if JS is the only thing that needs malloc, we need the deps_info mechanism, don't we?

Comment (sbc100) [reply src/library_pthread.js:665]: I did add this to deps info.  See the new entry for `pthread_create` below.   Not that its the use of `pthread_create` in users code that needs to go in deps_info not `__pthread_create_js` (which is called in directly from `pthread_create`.  Entries in deps_info always have to be symbols that exist in user object files.

Comment (sbc100) [reply src/library_pthread.js:665]: Also, note that I'm deliberately adding the dependency both here and in deps_info.  This one acts as a fail safe for cases where the deps_info for some reason didn't work.   If we don't add this one here then we get a runtime error instead of a build time error when deps_info isn't correct.

File: src/library_html5_webgl.js [hunk 2]
```diff
@@ -641,6 +638,10 @@ function handleWebGLProxying(funcs) {
 }
 
 handleWebGLProxying(LibraryHtml5WebGL);
+#endif // USE_PTHREADS
+
+#if LibraryManager.has('library_webgl.js')
+autoAddDeps(LibraryHtml5WebGL, '$GL');
```
Comment (kripken) [inline src/library_html5_webgl.js:644]: Should the comment on lines 74-78 be moved here? And, why is this change needed in this PR?

**Review by kripken — COMMENTED** [review]
Sounds good about deps_info, I forgot we add it in both places for better error messages.

File: src/library_html5_webgl.js (hunk 2 above)
Comment (sbc100) [reply src/library_html5_webgl.js:644]: Mostly for the deps_info test itself.    But actually I found it isn't actually exercised anymore so I can revert/split out this part of the change.

**Review by kripken — APPROVED** [review]
